### DISCUSSION TITLE
getting random primes of bitsize

### DISCUSSION THREAD

**Discussion by davejakenic** [Legrandin/pycryptodome#647] [Q&A]
*solved. Sorry

**Comment by James-E-A** [comment]
[![Comic, 1 panel. “Never have I felt so close to another soul; And yet so helplessly alone; As when I Google an error; And there's one result; A thread by someone with the same problem; And no answer; Last posted to in 2003”. A man stands in front of his desk, having risen so the chair has moved away behind him. He is holding on to his computer's screen, looking at it while visibly shaking the screen and shouting at it, “Who were you, DenverCoder9? <em>What did you see?!</em>”](https://imgs.xkcd.com/comics/wisdom_of_the_ancients.png)](https://www.explainxkcd.com/wiki/index.php/979%3A_Wisdom_of_the_Ancients)

[`Crypto.Math.Primality.generate_probable_prime(exact_bits=n)`](https://github.com/Legrandin/pycryptodome/blob/v3.17.0/lib/Crypto/Math/Primality.py#L280)

Note that this function is _only_ for cryptographic use, not for very small toy examples, as it will reject all `exact_bits <= 160`, with no allowed option to override.
